application/src/main/java/org/togetherjava/tjbot/features/moderation/history/PurgeHistoryCommand.java

```diff
@@ -0,0 +1,175 @@
+        event.deferReply(true).queue();
+
+        OptionMapping targetOption =
+                Objects.requireNonNull(event.getOption(USER_OPTION), "target is null");
+        Member author = Objects.requireNonNull(event.getMember(), "author is null");
+        String reason = Objects.requireNonNull(event.getOption(REASON_OPTION).getAsString(),
+                "reason is null");
+
+        OptionMapping durationMapping = event.getOption(DURATION);
+        int duration = durationMapping == null ? PURGE_MESSAGES_DEFAULT_DURATION
+                : durationMapping.getAsInt();
+
+        handleHistory(event, author, reason, targetOption, event.getHook(), duration);
+    }
+
+    private void handleHistory(SlashCommandInteractionEvent event, Member author, String reason,
+            OptionMapping targetOption, InteractionHook hook, int duration) {
+        Instant now = Instant.now();
+        Instant purgeMessagesAfter = now.minus(duration, ChronoUnit.HOURS);
+
+        User targetUser = targetOption.getAsUser();
+        String sourceChannelId = event.getChannel().getId();
+
+        if (!validateHierarchy(author, targetOption)) {
+            hook.sendMessage("Cannot purge history of user with a higher role than you").queue();
+            return;
+        }
+
+        List<String> messageIdsForDeletion = new ArrayList<>();
+
+        Stream<MessageHistoryRecord> fetchedMessageHistory =
+                database.writeAndProvide(context -> context.selectFrom(MESSAGE_HISTORY)
+                    .where(MESSAGE_HISTORY.AUTHOR_ID.equal(targetUser.getIdLong())
+                        .and(MESSAGE_HISTORY.CHANNEL_ID.equal(Long.valueOf(sourceChannelId)))
+                        .and(MESSAGE_HISTORY.SENT_AT.greaterOrEqual(purgeMessagesAfter)))
+                    .stream());
+
+        try (fetchedMessageHistory) {
+            fetchedMessageHistory.forEach(messageHistoryRecord -> {
+                String messageId = String.valueOf(messageHistoryRecord.getMessageId());
+                messageIdsForDeletion.add(messageId);
+                messageHistoryRecord.delete();
+
+                PurgeMessageListener.decrementRecordsCounterByOne();
+            });
+        } catch (DatabaseException exception) {
+            logger.error("unknown error during fetching message history records for {} command",
+                    COMMAND_NAME, exception);
+        }
+
+        handleDelete(messageIdsForDeletion, event.getChannel(), event.getHook(), targetUser, reason,
+                author, duration);
+    }
+
+    private void handleDelete(List<String> messageIdsForDeletion, MessageChannelUnion channel,
+            InteractionHook hook, User targetUser, String reason, Member author, int duration) {
+
+        if (messageIdsForDeletion.isEmpty()) {
+            handleEmptyMessageHistory(hook, targetUser, duration);
+            return;
+        }
+
+        if (hasSingleElement(messageIdsForDeletion)) {
```

Comment:
JDA docs suggest, if there's a single message it shouldnt be deleted using `deleteMessageById` instead of bulk deletion method(which can delete at least 2 or 100 tops in one batch).